app/controllers/course_user_data_controller.rb

```diff
@@ -190,20 +190,18 @@ def sudo
       redirect_to([@cud.course]) && return
     end
 
-    return unless request.post?
+    @users, @usersEncoded = @course.get_autocomplete_data
 
-    sudo_user = User.where(email: params[:sudo_email]).first
-    unless sudo_user
-      flash[:error] = "User #{params[:sudo_email]} does not exist"
-      redirect_to(action: :sudo) && return
-    end
+    return unless request.post?
 
-    sudo_cud = @course.course_user_data.where(user_id: sudo_user.id).first
+    sudo_cud = @course.course_user_data.where(id: params[:sudo_id]).first
```

Comment:
Previously
- User inputs `sudo_email`
- Search for corresponding user with that email, fail if not found
- Search for corresponding cud for that user, fail if not found
The new way is more direct
- User inputs `sudo_id`
- Search for corresponding cud for that id, fail if not found

Reply:
After that, we can obtain the corresponding user by finding the user with `id` equal to `sudo_cud.user_id`. Technically, we might needto check if `sudo_user` is non-nil. But practically, this shouldn't happen, and besides we wouldn't really be able to give a meaningful error message.